ui-tests/cypress/integration/content-types.spec.js

```diff
@@ -2,236 +2,290 @@ import {
   TEST_ID_PAGE_CONTAINER,
   TEST_ID_CONTENTTYPE_FORM,
 } from '../test-const/content-types-const';
+import { generateRandomId, generateRandomContentTypeCode } from '../support/utils';
+import HomePage from '../support/pageObjects/HomePage.js';
+
+describe('Content Types', () => {
+  let currentPage;
+  let contentTypeCode;
+  let contentTypeName;
 
-describe('Content Types - Complex', () => {
   beforeEach(() => {
-    cy.appBuilderLogin();
+    cy.kcLogin('admin').as('tokens');
+    cy.visit('/');
+
+    currentPage = new HomePage();
+    currentPage = currentPage.getMenu().getContent().open();
+    currentPage = currentPage.openTypes();
+
+    contentTypeCode = generateRandomContentTypeCode();
+    contentTypeName = generateRandomId();
   });
 
   afterEach(() => {
-    cy.appBuilderLogout();
+    cy.kcLogout();
   });
 
-  const contentTypeCode = 'BNR';
-  const TYPE_LIST = 'List';
-  const TYPE_MONOLIST = 'Monolist';
-  const TYPE_COMPOSITE = 'Composite';
-
-  const attributeCompositeTest = [
-    {
-      type: 'Hypertext',
-      codeValue: 'httext',
-      nameEnValue: 'Le Hyper Text',
-    }, {
-      type: 'Link',
-      codeValue: 'myLink',
-      nameEnValue: 'My Link',
-    }, {
-      type: 'Timestamp',
-      codeValue: 'currStamp',
-      nameEnValue: 'Curr Stamp',
-    },
-  ];
-
-  describe('Content Type Attribute - List', () => {
-    describe('nested attribute types that are not allowed in List attribute', () => {
-      it('attribute type selection should not contain Text, Longtext, Hypertext, Image, Attach, Link', () => {
-        cy.openContentTypeFormWith(contentTypeCode);
-        cy.addNewContentTypeAttribute(contentTypeCode, TYPE_LIST);
-        cy.getByName(TEST_ID_CONTENTTYPE_FORM.ATTRIBUTE_TYPE_DROPDOWN).should('not.contain', 'Text');
-        cy.getByName(TEST_ID_CONTENTTYPE_FORM.ATTRIBUTE_TYPE_DROPDOWN).should('not.contain', 'Longtext');
-        cy.getByName(TEST_ID_CONTENTTYPE_FORM.ATTRIBUTE_TYPE_DROPDOWN).should('not.contain', 'Hypertext');
-        cy.getByName(TEST_ID_CONTENTTYPE_FORM.ATTRIBUTE_TYPE_DROPDOWN).should('not.contain', 'Image');
-        cy.getByName(TEST_ID_CONTENTTYPE_FORM.ATTRIBUTE_TYPE_DROPDOWN).should('not.contain', 'Attach');
-        cy.getByName(TEST_ID_CONTENTTYPE_FORM.ATTRIBUTE_TYPE_DROPDOWN).should('not.contain', 'Link');
-      });
-    });
+  const addContentType = (code, name) => {
+    cy.log(`Add content type with code ${code}`);
+    currentPage = currentPage.getContent().addContentType();
```

Comment:
How can you be sure `currentPage` will have the correct value when this function is invoked?
It would be better to create the `currentPage` object here, or, at least, validate it is the correct one to perform the actions (maybe passing it as argument)

Reply:
Good point. Yeah I think it's better to move it outside and add the page object as parameter. Thanks!